pkg/restapi/issuer/operation/operations.go

```diff
@@ -794,6 +806,61 @@ func buildCredential(composeCredReq *ComposeCredentialRequest) (*verifiable.Cred
 	return credential, nil
 }
 
+func setContexts(credential *verifiable.Credential, credentialFormatOptions json.RawMessage) error {
+	if len(credentialFormatOptions) == 0 {
+		credential.Context = []string{"https://www.w3.org/2018/credentials/v1"}
+		return nil
+	}
+
+	var ctx struct {
+		Context interface{} `json:"@context,omitempty"`
+	}
+
+	err := json.Unmarshal(credentialFormatOptions, &ctx)
+	if err != nil {
+		return err
+	}
+
+	if ctx.Context == nil {
+		credential.Context = []string{"https://www.w3.org/2018/credentials/v1"}
+
+		return nil
+	}
+
+	credential.Context, credential.CustomContext, err = decodeContext(ctx.Context)
+
+	return err
+}
+
+// decodeContext decodes raw context(s).
+//
+// context can be defined as a single string value or array;
+// at the second case, the array can be a mix of string and object types
+// (objects can express context information); object context are
+// defined at the tail of the array.
+func decodeContext(c interface{}) ([]string, []interface{}, error) {
```

Comment:
can we move this function outside operation.go

Reply:
Sure

Reply:
moved this one including other private code from this package to internal.